### PR TITLE
Doc examples

### LINKED ISSUE #156: Update example pages in documentation because of new binder support
I think the examples needed more structure for some time. Now with the binder support, it seems like a good time to make these adjustments:

- https://pypsa.readthedocs.io/en/latest/examples.html
- https://pypsa.readthedocs.io/en/latest/quick_start.html

### LINKED ISSUE #222: Add example notebooks to documentation
In order to reframe the `pypsa.org` website, all examples that are linked to, should be added to the doc in a separate example section.

### PR DESCRIPTION
**Pull request by FabianHofmann**
Closes #222
Closes #156 

Also related to #93 (which could be tackled right away).

## Changes proposed in this Pull Request

Add example notebooks to the doc, see the current build [here](https://pypsa.readthedocs.io/en/doc-examples/). The question is: should we add all of them (I already kicked out two)? 
Secondly, I would like to have the example in appropriate categories, like 

**Basic**
**Power Flow Optimization**
**Sector-Coupling**
**Other**

Atm they are all together under the `Example` caption which seems a bit clumsy. 

So far, the notebook don't have any content. I'd say we add it as soon as we have the final arrangement (for sure we have to be aware of not putting to much pictures into it). Alternatively,  we can let readthedocs run the notebooks so we don't have to store any output in the repo. 

Nice thing of the setup, the example notebook directly link to the interactive binder page. 

# TODOs

- [x] Make a selection of notebooks (also all are possible)
- [x] Structure/Arrange all notebooks in subsection (optional) 
- [x] Update quickstart.rst and possibly other to not reference the `pypsa.org` example anymore but the readthedocs examples
- [ ] ...

After this PR is through the pypsa.org website could be updated to a more general overview website, about the whole pypsa project with pypsa, pypsa-eur, pypsa-eur-sec, atlite, technology-data, etc.

## Checklist

- [ ] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [ ] Unit tests for new features were added (if applicable).
- [ ] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).
- [ ] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.

### CODE REVIEW

**Comment by fneum** [comment]
This is great, especially the binder integration! From a first glance at the build:

- yes to categorisation of examples in subsections, the menu currently looks a bit crowded
- if readthedocs can run the notebooks, that would be preferable
- maybe one or two old examples can go, some may need a facelift, but generally keeping most of them would be preferable
- with the update of `pypsa.org` (yes!) the `website` folder should move to its own repository
- nice design :haircut:

**Comment by FabianHofmann** [comment]
Alright, ready for review @fneum ! It runs through with only 2 warnings (which I cannot resolve), before it were about 50 :-)

File: doc/examples-lopf.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#######################
+Power Flow Optimization
+#######################
+
+
+This section provides examples related to network optimization with PyPSA. Note that PyPSa allows for optimising networks in two ways, first using the ``pyomo`` interface and second using the in-home implementation (which is way faster and memory-efficient). Some newer examples use the in-house implementation while others are based in the pyomo package.
```
Comment (fneum) [inline doc/examples-lopf.rst:6]:
```suggestion
This section provides examples related to network optimization with PyPSA. Note that PyPSA allows for optimising networks in two ways, first using the ``pyomo`` interface and second using the in-home implementation (which is way faster and memory-efficient). Some newer examples use the in-house implementation while others are based in the pyomo package. 
```

Link to modules :mod:`pypsa.opf` and : mod:`pypsa.linopf`?

File: doc/examples/create-nblinks [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (fneum) [inline doc/examples/create-nblinks:1]: You run this manually, right?

File: doc/index.rst [hunk 3]
```diff
@@ -64,7 +64,6 @@ Documentation
 * :doc:`introduction`
 * :doc:`installation`
 * :doc:`quick_start`
-* :doc:`examples`
```
Comment (fneum) [inline doc/index.rst:67]: What would you think of the order

- getting started
- examples
- user guid
- help and references

?

File: doc/quick_start.rst [hunk 4]
```diff
@@ -4,172 +4,51 @@ Quick Start
...
-    print(network.storage_units_t.p)
+    network.add("Load", "My load",
+		bus="My bus 1",
+		p_set=100
+		q_set=100)
 
-    print(network.lines.s_nom_opt)
 
-    print(network.lines_t.p0)
+Note that fixed values of active and inactive power are set with ``p_set`` and ``q_set`` respectively. After initializing, there are plenty of options what you can do with your network. The examples section gives a broad overview.
```
Comment (fneum) [inline doc/quick_start.rst:54]: Link to examples section here?

**Review by fneum — CHANGES_REQUESTED** [review]
Really amazing to have the notebooks in the docs!

Could we maybe keep the minimal lopf example or merge it into the minimal pf example?

Only the outputs of scigrid-lopf-then-pf cell [18] look a bit funny

File: doc/examples/create-nblinks (hunk 2 above)
Comment (FabianHofmann) [reply doc/examples/create-nblinks:1]: yes

File: doc/index.rst (hunk 3 above)
Comment (FabianHofmann) [reply doc/index.rst:67]: yes, why not? that is probably more beginner friendly

File: doc/quick_start.rst (hunk 4 above)
Comment (FabianHofmann) [reply doc/quick_start.rst:54]: Actually, as far as I know there is no link option to a toctree header, one had to link to a specific notebook...

Comment (fneum) [reply doc/quick_start.rst:54]: Ah, I see. Not that important.

File: doc/index.rst (hunk 3 above)
Comment (fneum) [reply doc/index.rst:67]: Yes, that's what I thought

**Review by fneum — APPROVED** [review]

**Pull request merged**